packages/builder/src/pages/builder/index.svelte

```diff
@@ -1,11 +1,12 @@
 <script>
   import { redirect } from "@roxi/routify"
   import { auth } from "stores/portal"
+  import { sdk } from "@budibase/shared-core"
 
   auth.checkQueryString()
 
   $: {
-    if ($auth.user?.builder?.global) {
+    if (sdk.users.hasBuilderPermissions($auth.user)) {
```

Comment:
So, basically the difference is:
1. `isGlobalBuilder`, as you say its self explanatory, it is the current builder type user
2. `isBuilder`, checks if the user is a global builder, without an app ID it will fail for app builders, with an app ID then app builders can be permitted
3. `hasBuilderPermissions` - this means that the user could be a builder in an app, but you don't have the app ID to confirm which app it is, for example the decision to show them the builder portal. Its less meaningful in frontend, its used heavily in backend for requests that we aren't aware of an app for